### PR TITLE
Run Optical Flow on stream 0 when running driver > 460.

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michał Zientkiewicz <mzient@gmail.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It works around a bug #2912

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     * Run Optical Flow on stream 0 with problematic driver
 - Affected modules and functionalities:
     * Optical flow
 - Key points relevant for the review:
     * N/A
 - Validation and testing:
     * N/A
 - Documentation (including examples):
     * N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by mzient** [comment]
!build

File: dali/operators/sequence/optical_flow/optical_flow.cc [hunk 1]
```diff
@@ -81,6 +82,40 @@ constexpr int kNInputDims = 4;
...
+  // TODO(michalz): add an upper bound when the problem is fixed
+  cudaStream_t of_stream = ws.stream();
+#if NVML_ENABLED
+  {
+    static float driver_version = []()->float {
+      char version[80];
+      CUDA_CALL(nvmlInitChecked());
+      CUDA_CALL(nvmlSystemGetDriverVersion(version, sizeof version));
+
+      return std::stof(version);
```
Comment (jantonguirao) [inline dali/operators/sequence/optical_flow/optical_flow.cc:96]: Is it guaranteed that the version can always be converted to float? No alpha characters? Or double '.' in the version string?

File: dali/operators/sequence/optical_flow/optical_flow.cc [hunk 2]
```diff
@@ -81,6 +82,40 @@ constexpr int kNInputDims = 4;
...
+    }();
+    if (driver_version > 460)
+      of_stream = 0;
+  }
+#else
+  {
+    int driver_cuda_version = 0;
+    CUDA_CALL(cuDriverGetVersion(&driver_cuda_version));
+    if (driver_cuda_version >= 11030)
+      of_stream = 0;
```
Comment (jantonguirao) [inline dali/operators/sequence/optical_flow/optical_flow.cc:106]: Any reason to provide both this and the NVML implementation?

**Review by jantonguirao — APPROVED** [review]

File: dali/operators/sequence/optical_flow/optical_flow.cc (hunk 1 above)
Comment (mzient) [reply dali/operators/sequence/optical_flow/optical_flow.cc:96]: We already do it in two other places.

File: dali/operators/sequence/optical_flow/optical_flow.cc (hunk 2 above)
Comment (mzient) [reply dali/operators/sequence/optical_flow/optical_flow.cc:106]: With NVML we can distinguish driver versions, not CUDA versions - that's finer granularity, we can possibly pick up fixes earlier.

Comment (JanuszL) [reply dali/operators/sequence/optical_flow/optical_flow.cc:106]: How this work with 450.x driver and compatibility layer enabled? I think `cuDriverGetVersion` will report 11030.

Comment (mzient) [reply dali/operators/sequence/optical_flow/optical_flow.cc:106]: I have no idea, but that's for bare-metal builds without NVML, so I wouldn't sweat it too much.

Comment (JanuszL) [reply dali/operators/sequence/optical_flow/optical_flow.cc:106]: I think it will work as I said. 450.x will report 11030. Far from perfect TBH.

**Review by banasraf — APPROVED** [review]

**Review by JanuszL — APPROVED** [review]

**Pull request merged**